Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,65 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith(() =>
+                {
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+                    string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+                    return new FailReason($"Expected enum to equal {expectationName} by value{{reason}}, but found {subjectsName}.");
+                });
+        }
+
+        private static void HandleByName(IEquivalencyValidationContext context)
+        {
+            string subject = context.Subject.ToString();
+            string expected = context.Expectation.ToString();
+
+            Execute.Assertion
+                .ForCondition(subject == expected)
+                .FailWith(() =>
+                {
+                    decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+                    decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
```

Comment:
🙃 I would keep the subject-related lines together and then the expectation-related ones.

Reply:
If that's not a problem I will keep it like it is. I tried the other version but it looked more chaotic.